crates/sp-lightclient/src/lib.rs

```diff
@@ -510,24 +513,25 @@ impl<Header: HeaderT, Store: Storage<Header>> HeaderImporter<Header, Store> {
             Ok(SaltDerivationInfo {
                 eon_index: parent_salt_derivation_info.eon_index,
                 // if the salt is not revealed yet, check if salt will be revealed at this header for the current eon index
-                maybe_randomness: parent_salt_derivation_info.maybe_randomness.or(self
-                    .randomness_for_next_salt(
+                maybe_randomness: parent_salt_derivation_info.maybe_randomness.or(
+                    Self::randomness_for_next_salt(
+                        constants,
                         parent_salt_derivation_info.eon_index,
                         genesis_slot,
                         &pre_digest,
-                    )?),
+                    )?,
+                ),
             })
         }
     }
 
     /// Returns randomness used to derive the next salt if the next salt is revealed after importing header.
     fn randomness_for_next_salt(
-        &self,
+        constants: ChainConstants<Header>,
```

Comment:
Do you need owned value though? Using references will be much more efficient when compiled to WebAssembly AFAIK.